### PR TITLE
feat: restricted strategy list

### PR DESCRIPTION
**Pull request by 0xClandestine**

### CODE REVIEW

File: src/contracts/core/AllocationManager.sol
Comment (0xClandestine) [inline src/contracts/core/AllocationManager.sol:61]: Not a fan of repeat external calls to avsd, hoping we can merge these contracts.

File: src/contracts/core/AVSDirectory.sol
Comment (ypatil12) [inline src/contracts/core/AVSDirectory.sol:427]: Nit: can we save gas by not encoding every time?

File: src/contracts/core/AllocationManager.sol
Comment (ypatil12) [reply src/contracts/core/AllocationManager.sol:61]: It's possible we can merge the _new_ AVSD functionality (ie. opset functionality) into the ALM. Unsure about merging ALM into AVSD

File: src/contracts/core/AllocationManager.sol
Comment (ypatil12) [inline src/contracts/core/AllocationManager.sol:67]: Nit: comment isn't doing this check? I think you mean to put above `require(avsDirectory.isOperatorSlashable(params.operator, operatorSet), InvalidOperator());`

File: src/contracts/core/AllocationManager.sol
Comment (ypatil12) [inline src/contracts/core/AllocationManager.sol:174]: The net effect here is that you should ONLY be able to allocate to strategies that are whitelisted but can deallocate regardless. I think this is great. Can we add this assumption to the interface?

File: src/contracts/interfaces/IAVSDirectory.sol
Comment (ypatil12) [inline src/contracts/interfaces/IAVSDirectory.sol:329]: Nit: should we natspec the params?

File: src/test/unit/AllocationManagerUnit.t.sol
Comment (ypatil12) [inline src/test/unit/AllocationManagerUnit.t.sol:114]: Can we add three tests: 

1. Revert when you try to slash for a strategy that does not exist. The operator should have allocated to the strategy, but then the AVS removes the strategy from their list of strategies we should revert the slash
2. Revert when the operator tries to allocate for a strategy not added to the operatorSet
3. Validate that we can deallocate, even if the strategy is not part of the operator set

File: src/contracts/core/AVSDirectory.sol
Comment (ypatil12) [inline src/contracts/core/AVSDirectory.sol:427]: Can we add tests to the AVSD to validate this is calculated correctly? Just sanity checking the output from those fuzz tests

**Review by ypatil12 — COMMENTED** [review]
Mainly want some more test coverage

File: src/contracts/core/AVSDirectory.sol
Comment (0xClandestine) [reply src/contracts/core/AVSDirectory.sol:427]: noted, will fix if we don't move new avsd logic.

File: src/contracts/core/AllocationManager.sol
Comment (0xClandestine) [reply src/contracts/core/AllocationManager.sol:174]: You mean document it?

Comment (ypatil12) [reply src/contracts/core/AllocationManager.sol:173]: Ya in the interface comments

File: src/test/unit/AllocationManagerUnit.t.sol
Comment (ypatil12) [reply src/test/unit/AllocationManagerUnit.t.sol:114]: I think we're still missing these?

**Review by ypatil12 — APPROVED** [review]

File: src/contracts/core/AllocationManager.sol
Comment (wadealexc) [inline src/contracts/core/AllocationManager.sol:62]: Can we not have this method accept an operator set as an argument and fetch strategies from the AVSD? That seems far preferable to passing in an array of strategies here.

File: src/contracts/core/AllocationManager.sol
Comment (wadealexc) [inline src/contracts/core/AllocationManager.sol:63]: `Asser` -> `Assert`

File: src/contracts/core/AllocationManager.sol
Comment (0xClandestine) [reply src/contracts/core/AllocationManager.sol:62]: FIxed in #855, but the fix introduces a via-ir compile error. Thinking we wait until #845 is merged which should reduce code complexity.

**Pull request closed without merging**
